### PR TITLE
Update setup

### PR DESCRIPTION
**Pull request by byronz**
This PR improves the requirement*.txt in dash repo, it unifies the requirements using the python version condition, and lists all dependencies in txt file instead of listing them in setup.py; this can give docker setup more flexibility. 

## Contributor Checklist

- [x] I have broken down my PR scope into the following TODO tasks
   -  [x] improve the requirements.txt in setup.py 
   -  [x] move the `dash.testing` dependencies into main install
- [x] I have run the tests locally and they passed. (refer to testing section in [contributing](../CONTRIBUTING.md))
- [ ] I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR

### optionals

- [ ] I have added entry in the `CHANGELOG.md`
- [ ] If this PR needs a follow-up in **dash docs**, **community thread**, I have mentioned the relevant URLS as follow
    -  [ ] this github [#PR number]() updates the dash docs
    -  [ ] here is the show and tell thread in plotly dash community

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -2,65 +2,63 @@
...
-        'plotly',
-        'dash_renderer==0.24.0',
-        'dash-core-components==0.48.0',
-        'dash-html-components==0.16.0',
-        'dash-table==3.7.0'
-    ],
+    long_description=io.open("README.md", encoding="utf-8").read(),
+    long_description_content_type="text/markdown",
+    install_requires=read_req_file("install"),
+    extra_require={"intg": read_req_file("intg")},
```
Comment (byronz) [inline setup.py]: each key here can be used by `pip install -e .[key]`

File: requires-intg.txt [hunk 2]
```diff
@@ -0,0 +1,9 @@
+dash_flow_example==0.0.5
+dash-dangerously-set-inner-html
+six
+mock
+flake8
+pylint==1.9.4;python_version<"3.7"
+pylint==2.3.1;python_version=="3.7"
+astroid==2.2.5;python_version=="3.7"
```
Comment (byronz) [inline requires-intg.txt]: this can avoid having two requirements per python version

File: requires-ci.txt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+dash_flow_example==0.0.5
+dash-dangerously-set-inner-html
+six
+mock
+flake8
+pylint==1.9.4;python_version<"3.7"
+pylint==2.3.1;python_version=="3.7"
+astroid==2.2.5;python_version=="3.7"
```
Comment (alexcjohnson) [inline requires-ci.txt]: didn't know you could do that, cool! Would it be better to say `>="3.7"`? NBD, I guess we'll know pretty quickly if we move to 3.8 and things break...

**Comment by byronz** [comment]
> Not sure what's going on with py2.7, but this is a nice simplification so if you can get it working I'm all for it! 💃

looks like a cache key issue, I will add python version as well

**Pull request merged**